src/app/components/navbar/Navbar.jsx

```diff
@@ -1,54 +1,31 @@
-import React, { Component } from "react";
+import React from "react";
 import { Link } from "react-router";
-import { connect } from "react-redux";
 import PropTypes from "prop-types";
+import url from "../../utilities/url";
+import cookies from "../../utilities/cookies";
+import auth from "../../utilities/auth";
+import user from "../../utilities/api-clients/user";
+import PreviewNav from "./preview-nav";
+
+const Navbar = ({ user, rootPath, workingOn, config, location }) => {
+    const authenticated = auth.isAuthenticated(user) || false;
+    const routeIsACollectionPage = path => {
+        return path.indexOf(`/collections`) >= 0;
+    };
 
-import url from "../utilities/url";
-import cookies from "../utilities/cookies";
-import auth from "../utilities/auth";
-import PreviewNav from "./PreviewNav";
-import user from "../utilities/api-clients/user";
-
-const propTypes = {
-    config: PropTypes.shape({
-        enableDatasetImport: PropTypes.bool
-    }),
-    user: PropTypes.object.isRequired,
-    workingOn: PropTypes.shape({
-        id: PropTypes.string.isRequired,
-        name: PropTypes.string
-    }),
-    rootPath: PropTypes.string.isRequired,
-    location: PropTypes.object.isRequired,
-    dispatch: PropTypes.func.isRequired
-};
-
-export class NavBar extends Component {
-    constructor(props) {
-        super(props);
-
-        this.handleLogoutClick = this.handleLogoutClick.bind(this);
-    }
-
-    handleLogoutClick() {
+    const handleLogoutClick = () => {
         user.logOut();
-    }
-
-    renderWorkingOnItem() {
-        const workingOn = this.props.workingOn || {};
-        const showWorkingOn = workingOn.id;
-        if (!showWorkingOn) {
-            return;
-        }
+    };
 
-        const path = this.props.location.pathname;
-        if (this.routeIsACollectionPage(path)) {
+    const renderWorkingOnItem = workingOn => {
+        if (!workingOn) return null;
+        if (routeIsACollectionPage(location.pathname)) {
```

Comment:
How are we able to use location? Does line 20 not need to be renderWorkingOnItem = (workingOn, location) => {

Reply:
again, location is part of react router. it comes form wrapping those components in router :-)